cogs/respects/core.py

```diff
@@ -89,20 +100,20 @@ async def checkLastRespect(self, ctx: Context):
                         await chConfig.clear()
                         return False
 
-        confMsgsBetween = await guildConfig.get_attr(KEY_MSGS_BETWEEN)()
-        confTimeBetween = await guildConfig.get_attr(KEY_TIME_BETWEEN)()
-        oldRespectTime = await chConfig.get_attr(KEY_TIME)()
+        confMsgsBetween: int = await guildConfig.get_attr(KEY_MSGS_BETWEEN)()
+        confTimeBetween: float = await guildConfig.get_attr(KEY_TIME_BETWEEN)()
+        oldRespectTime: Optional[float] = await chConfig.get_attr(KEY_TIME)()
```

Comment:
This being `None`-able may trigger type checkers in the code portion below because `datetime.fromtimestamp` doesn't accept `None`. Can you add a `None` check and return `False` if `oldRespectTime` is `None`, plus a warning log line? We will categorize this as _the last respect could not be retrieved_ as documented in the method docstring.
That `None` check can tell static type checkers that `oldRespectTime` is `float` after the check.

Since this PR is about adding type hints, I suppose we can make this change in a separate PR, but it's small enough so I don't mind if you want to add it in this PR.

Reply:
Good catch, this won't be `None` at runtime.